classes/package.php

```diff
@@ -299,29 +330,42 @@ public static function get_record_by_hash(string $hash): ?array {
      *
      * @param array|null $conditions
      * @return array
-     * @throws dml_exception
      * @throws moodle_exception
      */
     public static function get_records(?array $conditions = null): array {
         global $DB;
-        $records = $DB->get_records('qtype_questionpy_package', $conditions);
+        $pkgversions = $DB->get_records('qtype_questionpy_pkgversion', $conditions);
         $packages = array();
-        foreach ($records as $package) {
-            $package = (array)$package;
-            [$language, $name, $description] = self::get_languagedata($package["id"]);
-            $tags = self::get_tagdata($package["id"]);
-            $temp = [
-                'languages' => $language,
-                'name' => $name,
-                'description' => $description,
-                'tags' => $tags
-            ];
-            $package = array_merge($package, $temp);
+        foreach ($pkgversions as $pkgversion) {
+            $package = self::get_package_data($pkgversion->packageid);
+            $package = array_merge((array) $pkgversion, (array) $package);
+
             $packages[] = array_converter::from_array(self::class, $package);
         }
         return $packages;
     }
 
+    /**
+     * Get package related data like name, description and tags.
+     *
+     * @param string $packageid
+     * @return mixed|\stdClass
+     * @throws moodle_exception
+     */
+    private static function get_package_data(string $packageid) {
```

Comment:
`$DB->get_record` & co sind da nicht ganz so gut dokumentiert, aber da diese Methode davon ausgeht, dass da Objekte zurückkommen, sollten wir sie auch als solches deklarieren:
```suggestion
    /**
     * Get package related data like name, description and tags.
     *
     * @param string $packageid
     * @return object
     * @throws moodle_exception
     */
    private static function get_package_data(string $packageid): object {
```